lib/container_server.go

```diff
@@ -756,17 +770,20 @@ func (c *ContainerServer) HasSandbox(id string) bool {
 }
 
 // RemoveSandbox removes a sandbox from the state store
-func (c *ContainerServer) RemoveSandbox(id string) {
+func (c *ContainerServer) RemoveSandbox(id string) error {
 	sb := c.state.sandboxes.Get(id)
 	if sb == nil {
-		return
+		return nil
 	}
 
 	c.stateLock.Lock()
-	c.removeSandboxPlatform(sb)
+	if err := c.removeSandboxPlatform(sb); err != nil {
+		return err
```

Comment:
same issue here

Reply:
I added it as `defer`.

Reply:
I am a little worried about adding defers,  Wern't these causing performance issues? Or was that in libpod.

Reply:
Defer is causing a significant higher performance cost than not using it, yes. If this is a problem I can change it again.

Reply:
But we use it for other methods of the `ContainerServer` too, so it should not be that harmful in our case.

Reply:
@rhatdan was the issue that we were holding the lock more than it was necessary?  I'd expect most of the cost to come from there